src/PKSim.Core/Services/HIDiseaseStateImplementation.cs

```diff
@@ -0,0 +1,220 @@
+         var portalOrgans = new[] { STOMACH, SMALL_INTESTINE, LARGE_INTESTINE, SPLEEN, PANCREAS };
+         portalOrgans.Each(x => updateBloodFlow(x, _portalFlowScalingFactor[score]));
+
+         //Hepatic
+         updateBloodFlow(LIVER, _hepaticFlowScalingFactor[score]);
+
+         //Renal
+         updateBloodFlow(KIDNEY, _renalFlowScalingFactor[score]);
+
+         //other organs
+         var otherOrgans = new[] { BONE, FAT, GONADS, HEART, MUSCLE, SKIN };
+         otherOrgans.Each(x => updateBloodFlow(x, _otherOrgansFlowScalingFactor[score]));
+      }
+
+      private void updateVolumes(Individual individual, Action<IParameter, double, bool> updateParameterFunc)
+      {
+         var score = childPughScoreFor(individual);
+         var organism = individual.Organism;
+         var parameter = organism.Container(LIVER).Parameter(Constants.Parameters.VOLUME);
+         updateParameterFunc(parameter, _hepaticVolumeScalingFactor[score], true);
+      }
+
+      private Action<string, double> updateBloodFlowDef(Action<IParameter, double, bool> updateParameterFunc, IContainer organism) => (organName, factor) =>
+      {
+         var parameter = organism.Container(organName).Parameter(SPECIFIC_BLOOD_FLOW_RATE);
+         updateParameterFunc(parameter, factor, true);
+      };
+
+      private double childPughScoreFor(Individual individual)
+      {
+         return individual.OriginData.DiseaseStateParameters.FindByName(CHILD_PUGH_SCORE).Value;
+      }
+
+      public bool ApplyForPopulationTo(Individual individual)
+      {
+         throw new NotImplementedException();
+      }
+
+      public Individual CreateBaseIndividualForPopulation(Individual originalIndividual)
+      {
+         throw new NotImplementedException();
+      }
+
+      public void ResetParametersAfterPopulationIteration(Individual individual)
+      {
+         throw new NotImplementedException();
+      }
+
+      private void updateParameterValue(IParameter parameter, double value, bool useAsFactor)
+      {
+         var valueToUse = useAsFactor ? parameter.Value * value : parameter.Value;
+         parameter.Value = valueToUse;
+      }
+
+      private void updateParameter(IParameter parameter, double value, bool useAsFactor)
+      {
+         var valueToUse = useAsFactor ? parameter.Value * value : parameter.Value;
+         updateValueOriginsFor(parameter);
+         if (parameter is IDistributedParameter distributedParameter)
+         {
+            distributedParameter.ScaleDistributionBasedOn(valueToUse / distributedParameter.Value);
+            return;
+         }
+
+         //We are using a formula, we override with a constant
+         if (parameter.Formula.IsExplicit())
+         {
+            parameter.Formula = _formulaFactory.ConstantFormula(valueToUse, parameter.Dimension);
+            //Make sure the formula is indeed used in case the value was overwritten before as fixed value
+            parameter.IsFixedValue = false;
```

Comment:
@Yuri05 this is a bug in the other module BTW. I will make this code generic so all good. (and it was not sued I think)